3.2/root/usr/share/container-scripts/mongodb/init-petset-replset.sh

```diff
@@ -118,31 +70,28 @@ function add_member() {
   local replset_addr
   replset_addr="${MONGODB_REPLICA_NAME}/${endpoints}"
 
-  if ! mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host "${replset_addr}" --eval "${script}" --quiet; then
+  if ! mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host "${replset_addr}" --eval "while (!rs.add('${host}').ok) { sleep(100); }" --quiet; then
     info "ERROR: couldn't add host to replica set!"
     return 1
   fi
 
-  info "Successfully added to replica set"
   info "Waiting for PRIMARY/SECONDARY status ..."
 
-  local rs_status_out
-  rs_status_out="$(mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host "${replset_addr}" --eval "${WAIT_PRIMARY_OR_SECONDARY_STATUS}" --quiet || :)"
-
-  if ! echo "${rs_status_out}" | grep -xqs '\(SECONDARY\|PRIMARY\)'; then
-    info "ERROR: failed waiting for PRIMARY/SECONDARY status. Command output was:"
-    echo "${rs_status_out}"
-    echo "==> End of the error output <=="
-    return 1
-  fi
+  while [[ "$(mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --eval 'rs.status().myState' --quiet || :)" != "2" ]]; do
```

Comment:
No really sure why `WAIT_PRIMARY_STATUS` contained checking that `rs.status()` contains `myState` property. Are you sure, that `myState` is in the output also early after adding member to replicaset?

Maybe using `rs.isMaster()` would be safer.